gobblin-metastore/src/main/java/org/apache/gobblin/metastore/MysqlJobStatusStateStore.java

```diff
@@ -55,4 +59,20 @@ public MysqlJobStatusStateStore(DataSource dataSource, String stateStoreTableNam
     return getAll(storeName, flowExecutionId + "%", true);
   }
 
+  @Override
+  public void put(String storeName, String tableName, State state) throws IOException {
```

Comment:
I think the read-modify-write pattern for put() should be ok - since we are only tracking 5 events per job. Metrics would be helpful though - e.g. total time spent in put().